### PR TITLE
Implement OcExpandingDropdown | Redesign ApplicationMenu | Redesign Topbar dropdowns

### PR DESCRIPTION
**Pull request by lookacat**
## Description
I would explain things to a potential reviewer on how and why :)

## Screenshots (outdated):
captures_chrome-capture-2023-2-10 (1)

[screen-capture (4).webm](https://user-images.githubusercontent.com/786551/225199173-0193f148-ef57-4137-b448-bcb0a7f973f2.webm)

The component can be used in different cases.
With `expand-head: false` and no `head` slot

[screen-capture (2).webm](https://user-images.githubusercontent.com/786551/224301063-70fd3c69-251e-46d6-98a5-f1592c82929e.webm)

With `expand-head: true` and the owncloud logo in the `head` slot

[screen-capture (3).webm](https://user-images.githubusercontent.com/786551/224427385-7ecc6378-7839-411f-bbae-b0da16372c2d.webm)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/33562/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2023-3-10-11-57-01.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/copy-and-move-resources-in-personal-space-alice-2023-3-10-11-58-10.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/file-alice-2023-3-11-12-06-14.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/folder-alice-2023-3-11-12-05-05.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/public-link-alice-2023-3-10-11-59-17.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/quick-link-alice-2023-3-11-12-00-24.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/re-sharing-brian-2023-3-11-12-02-46.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/search-in-personal-spaces-brian-2023-3-11-12-03-56.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/upload-resources-in-personal-space-alice-2023-3-11-12-07-20.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33562/tracing/user-should-be-able-to-read-and-dismiss-notifications-alice-2023-3-11-12-01-33.zip 


</p></details>
:boom: The e2e-oc10 tests pipeline failed. The build has been cancelled.

File: packages/web-runtime/src/mixins/navigationMixin.ts
Comment (kulmann) [inline packages/web-runtime/src/mixins/navigationMixin.ts:1]: This file has been removed in master recently, see https://github.com/owncloud/web/pull/8681
Is this a case of rebase-gone-wrong? ;-)

File: packages/web-runtime/src/components/Topbar/UserMenu.vue [hunk 1]
```diff
@@ -174,39 +173,43 @@ export default defineComponent({
 </script>
 
 <style lang="scss" scoped>
-.user-menu-list li {
-  align-items: center;
-  display: flex;
-  margin: var(--oc-space-xsmall) 0;
+.user-menu-list {
+  padding: 10px;
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/UserMenu.vue:177]: Please use `--oc-space-...` vars whenever possible. If you need exactly 10 pixels you can use `calc` to add 2px to an existing var... but would prefer to stay in the `8px` sizing base.

File: packages/web-runtime/src/components/Topbar/TopBar.vue [hunk 2]
```diff
@@ -216,11 +217,8 @@ export default {
   }
 
   .oc-topbar-left {
-    gap: 30px;
-
-    img.oc-logo-image {
-      height: 38px;
-    }
+    width: 10rem;
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/TopBar.vue:220]: Why do you need the `width`? Value seems odd..

File: packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue [hunk 3]
```diff
@@ -4,55 +4,58 @@
...
-        </li>
-      </oc-list>
-    </oc-drop>
+      <template #toggle>
+        <oc-icon name="grid" size="large" class="oc-flex" />
+      </template>
+      <template #head="{ dropdownVisible }">
+        <div class="dropdown-title-wrapper">
+          <transition name="slide-fade">
+            <h3 v-if="dropdownVisible">Applications</h3>
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue:19]: two things:
- string is not translated
- `h3` is probably not correct here. If I see it correctly then there is no `h2` in the hierarchy (`h1` should be there), so you'd need to use a `h2` tag.

File: packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue [hunk 4]
```diff
@@ -87,46 +101,123 @@ export default defineComponent({
...
+        position: absolute;
+        right: calc(1rem + 5px);
+      }
     }
   }
 }
+
+/* Application-Gradients */
+// TODO: Add default gradient / or add prop to choose gradient in the future
+.gradient-files {
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue:184]: Having dedicated classes for the different apps is not an option. Doesn't scale. Also pretty difficult in theming.

File: packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue [hunk 5]
```diff
@@ -4,55 +4,58 @@
...
+              :href="n.url"
+              :to="n.path"
+              appearance="raw"
+              :class="[`${n.gradient}`, { 'router-link-active': n.active }]"
+              variation="inverse"
+            >
+              <span class="icon-box">
+                <oc-icon size="medium" :name="n.icon" />
+              </span>
+              <span v-text="$gettext(n.title.slice(0, 5))" />
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue:39]: ???

File: packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue:1]: Why / how did you decide to go with a `tile` look and feel for the apps? I'm not a fan to be honest. Also, as you already hacked around it, it creates followup issues with the app name length.

**Review by kulmann — CHANGES_REQUESTED** [review]
I was under the assumption that you're working on:

Screenshot 2023-03-27 at 21 43 10

So I was very surprised to find this:

Screenshot 2023-03-27 at 21 44 00

Before you continue any work on this PR, please fill in details to https://github.com/owncloud/web/issues/8691 and find agreement with @tbsbdr - until now that issue has only placeholder content and is not justification for this PR. 😞

File: packages/web-runtime/src/mixins/navigationMixin.ts
Comment (lookacat) [reply packages/web-runtime/src/mixins/navigationMixin.ts:1]: yes

File: packages/web-runtime/src/components/Topbar/TopBar.vue (hunk 2 above)
Comment (lookacat) [reply packages/web-runtime/src/components/Topbar/TopBar.vue:220]: i think its dev leftover

File: packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue
Comment (lookacat) [reply packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue:1]: I can revert it, thought it was an improvement. Again lightmode is broken after rebase maybe try darkmode.

**Comment by hurradieweltgehtunter** [comment]
Thanks @lookacat for including me. I checked your proposal and cool that you are working on this. I also had a few ideas for improvement of the app switcher in the past.

Here are a few thoughts:
- Is there an underlying UX concept for this re-design? My impression is, that there is none. I would start here first or go back to it, especially since the app switcher is a very basic and always visible element. It will become more and more important in the future (maybe not now) when more apps get added to the OC universe.
- I'm not really in favor of the tiles view as the labels are exceeding the space – like you already encountered :) I think a list view is more suitable. Have a look at OneDrive, I like their app switcher and I think we can learn from it because they already have many apps:

grafik

- I'm also not in favor of the hovering effect of the tiles and the app switcher icon. I can not remember if we use the hover scaling anywhere else. We should stay consistent.
- I also would like it if we would design the whole selection box like a canvas element floating above the browser view. From a UX perspective, the app switcher is one layer above the current browser view (which represent one specific apps` content).


FYI: I got a few conflicts when building tokens. Not sure how relevant that is.

```
Property Value Collisions:
Collision detected at: font.weight.light! Original value: 300, New value: 300
Collision detected at: font.weight.light! Original value: /Users/flo/Desktop/oc-web/packages/design-system/src/tokens/docs/font.yaml, New value: /Users/flo/Desktop/oc-web/packages/design-system/src/tokens/ods/font.yaml
Collision detected at: font.weight.light! Original value: true, New value: true
Collision detected at: font.weight.normal! Original value: 400, New value: 400
Collision detected at: font.weight.normal! Original value: /Users/flo/Desktop/oc-web/packages/design-system/src/tokens/docs/font.yaml, New value: /Users/flo/Desktop/oc-web/packages/design-system/src/tokens/ods/font.yaml
Collision detected at: font.weight.normal! Original value: true, New value: true
Collision detected at: font.weight.bold! Original value: 700, New value: 700
Collision detected at: font.weight.bold! Original value: /Users/flo/Desktop/oc-web/packages/design-system/src/tokens/docs/font.yaml, New value: /Users/flo/Desktop/oc-web/packages/design-system/src/tokens/ods/font.yaml
Collision detected at: font.weight.bold! Original value: true, New value: true
```

**Comment by kulmann** [comment]
since this PR went into a direction that we decided against (tiles for the app switcher) I'll close this PR now. Thanks for your efforts @lookacat

**Pull request closed without merging**